api/src/pcapi/routes/public/individual_offers/v1/serialization.py

```diff
@@ -943,3 +947,25 @@ class Config:
 
 class GetOfferersVenuesQuery(serialization.ConfiguredBaseModel):
     siren: str | None = pydantic.Field(example="123456789")
+
+
+class GetProductsListByEansQuery(serialization.ConfiguredBaseModel):
+    eans: str | None = pydantic.Field(example="0123456789123,0123456789124")
+
+    @pydantic.validator("eans")
+    def validate_ean_list(cls, eans: str) -> list:
+        """The ean list must contain at least one element, at most 100
+        An ean must be a 13 digit integer"""
+        ean_list = eans.split(",")
+        if len(ean_list) > 100:
+            raise ValueError("Too many EANs")
+        if len(ean_list) == 0:
+            raise ValueError("EAN list must not be empty")
+        for ean in ean_list:
+            if not str(ean).isdigit():
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏Du coup si on attend une liste de `str`, pourquoi on cast ici ?

Reply:
pas besoin en effet